controllers/cruisecontroltask_types.go

```diff
@@ -62,71 +52,112 @@ func (t *CruiseControlTask) Apply(instance *kafkav1beta1.KafkaCluster) {
 	if t == nil || instance == nil {
 		return
 	}
-
+	//nolint:exhaustive
 	switch t.Operation {
-	case OperationAddBroker, OperationRemoveBroker:
+	case kafkav1alpha1.OperationAddBroker, kafkav1alpha1.OperationRemoveBroker:
 		if state, ok := instance.Status.BrokersState[t.BrokerID]; ok {
 			state.GracefulActionState.CruiseControlState = t.BrokerState
-			state.GracefulActionState.CruiseControlTaskId = t.TaskID
-			state.GracefulActionState.TaskStarted = t.StartedAt
-			state.GracefulActionState.ErrorMessage = t.Err
+			state.GracefulActionState.CruiseControlOperationReference = t.CruiseControlOperationReference
 			instance.Status.BrokersState[t.BrokerID] = state
 		}
-	case OperationRebalanceDisks:
+	case kafkav1alpha1.OperationRebalance:
 		if state, ok := instance.Status.BrokersState[t.BrokerID]; ok {
 			if volState, ok := state.GracefulActionState.VolumeStates[t.Volume]; ok {
 				volState.CruiseControlVolumeState = t.VolumeState
-				volState.CruiseControlTaskId = t.TaskID
-				volState.TaskStarted = t.StartedAt
-				volState.ErrorMessage = t.Err
+				volState.CruiseControlOperationReference = t.CruiseControlOperationReference
 				instance.Status.BrokersState[t.BrokerID].GracefulActionState.VolumeStates[t.Volume] = volState
 			}
 		}
 	}
 }
 
+func (t *CruiseControlTask) SetCruiseControlOperationRef(ref corev1.LocalObjectReference) {
+	if t == nil {
+		return
+	}
+	t.CruiseControlOperationReference = &ref
+}
+
+func (t *CruiseControlTask) SetStateScheduled() {
+	//nolint:exhaustive
```

Comment:
**Recommended**: We **SHOULD** also leave a comment here why it is required to override the linter for future readers.

Example:
```suggestion
	// nolint:exhaustive // Note: Not all CC operations require updates to the CruiseControlTask state.
```